Assets/Scripts/TerrainDestroyer.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+
+public class TerrainDestroyer : MonoBehaviour
+{
+
+    // Tilemaps needed to draw/erase tiles from them map
+    public Tilemap terrain;
+    public Tilemap backgroundTerrain;
+
+    // Tile that gets drawn in the background when tiles are destroyed
+    public TileBase explodedTile;
+
+    // Singleton instance
+    public static TerrainDestroyer instance = null;
+
+    private void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+        }
+        else if (instance != null)
+        {
+            Destroy(gameObject);
+        }
+
+    }
+
+    // Caculates the distance from the explosion to each potential from that center
+    // It then destroys any tile that is within the the radius of the explosion
+    public void DestroyTerrain(Vector3 explosionLocation, float radius)
+    {
+
+        /// The step function moves up by a small step because that is the scale of the tile map and we need to check every tile for this to work
+        for (float x = -radius; x < radius; x += 0.05f)
```

Comment:
Probably moving this from a magic number to a public float somewhere? In case we want to change this at a certain time? Or even better, if it is in any way possible to pick this up from the game object of the tile (if that is a parameter that is a function from the size of the tilemap or something like that). Just in case we decided to change the scale.

Reply:
Yes, we will want to look at this moving forward. Maybe I can take a look with the "interaction with loot" task